File: backend/app/schemas/telegram.py
import uuid
from typing import Any
from pydantic import BaseModel


class TelegramSettingsUpdate(BaseModel):
    is_enabled: bool | None = None
    notification_events: dict[str, Any] | None = None


class TelegramSettingsResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    company_id: uuid.UUID
    is_enabled: bool
    notification_events: dict[str, Any]


class TelegramConnectResponse(BaseModel):
    link_token: str
    bot_username: str | None
    instructions: str

File: frontend/src/types/telegram.ts
export interface TelegramSettings {
  id: string
  company_id: string
  is_enabled: boolean
  notification_events: Record<string, boolean>
}

export interface TelegramSettingsUpdate {
  is_enabled?: boolean
  notification_events?: Record<string, boolean>
}

export interface TelegramConnectResponse {
  link_token: string
  bot_username: string | null
  instructions: string
}
